ddtrace/contrib/django/utils.py

```diff
@@ -85,10 +87,25 @@ def get_request_uri(request):
             log.debug("Failed to build Django request host", exc_info=True)
             host = "unknown"
 
+    # If request scheme is missing, possible in case where wsgi.url_scheme
+    # environ has not been set, return None and skip providing a uri
+    if request.scheme is None:
+        return
+
     # Build request url from the information available
     # DEV: We are explicitly omitting query strings since they may contain sensitive information
     urlparts = dict(scheme=request.scheme, netloc=host, path=request.path, params=None, query=None, fragment=None)
 
+    # If any url part is a SimpleLazyObject, use its __class__ property to cast
+    # str/bytes and allow for _setup() to execute
+    for (k, v) in urlparts.items():
+        if isinstance(v, SimpleLazyObject):
+            if issubclass(v.__class__, str):
+                v = str(v)
+            elif issubclass(v.__class__, bytes):
+                v = bytes(v)
```

Comment:
what about the `else` case here (when v is an instance of SimpleLazyObject, but not a subclass of str or bytes)? We're going to end up adding a SimplyLazyObject to urlparts, is that what we want?

Reply:
good point, I changed it to handle `bytes` otherwise cast to `str`.

Reply:
is it worth using the value of `None` in the else case rather than assuming it's stringable?

Reply:
A case where we should be more defensive. If the lazy object is not `str` or `bytes`, we should skip it building a uri here then if we want to avoid casting another type.